### ISSUE TITLE
StackStorm/hubot-slack does not function with latest hubot

### ISSUE THREAD

**Issue by JohnWelborn** [StackStorm/st2#2896]
I'd like to run the latest version of hubot with stackstorm integration.

When I install hubot following the instructions on https://github.com/StackStorm/hubot-slack/ . I end up with a mostly functional bot+integration but it uses the upstream hubot-slack version that does not support the use of slack-attachment as called by hubot-stackstorm's post_data.js for when a message contains text after a {~}, such as "first{~}second"

When I force npm to install StackStorm/hubot-slack with the latest hubot, I get the following error:

```
[Mon Sep 12 2016 20:56:28 GMT+0000 (UTC)] DEBUG Chatops message received: {"trace_context": {"id_": "57d6d693726e9c5c59fabe1c", "trace_tag": null}, "payload": {"whisper": false, "message": "test6{~}", "user": null, "channel": "XXXXXXXXX", "extra": null}}
[Mon Sep 12 2016 20:56:28 GMT+0000 (UTC)] DEBUG Sending to XXXXXXXXX: test6
[Mon Sep 12 2016 20:56:28 GMT+0000 (UTC)] ERROR TypeError: Cannot read property 'send' of undefined
  at SlackBot.send (/opt/hubot/node_modules/hubot-slack/src/slack.coffee:209:16)
  at Robot.messageRoom (/opt/hubot/node_modules/hubot/src/robot.coffee:562:5)
  at SlackDataPostHandler.postData (/usr/lib/node_modules/hubot-stackstorm/lib/post_data.js:79:28)
  at EventSource.<anonymous> (/usr/lib/node_modules/hubot-stackstorm/scripts/stackstorm.js:383:25)
  at emitOne (events.js:77:13)
  at EventSource.emit (events.js:169:7)
  at _emit (/usr/lib/node_modules/hubot-stackstorm/node_modules/eventsource/lib/eventsource.js:172:17)
  at parseEventStreamLine (/usr/lib/node_modules/hubot-stackstorm/node_modules/eventsource/lib/eventsource.js:186:9)
  at IncomingMessage.<anonymous> (/usr/lib/node_modules/hubot-stackstorm/node_modules/eventsource/lib/eventsource.js:150:11)
  at emitOne (events.js:77:13)
  at IncomingMessage.emit (events.js:169:7)
  at readableAddChunk (_stream_readable.js:153:18)
  at IncomingMessage.Readable.push (_stream_readable.js:111:10)
  at HTTPParser.parserOnBody (_http_common.js:124:22)
  at TLSSocket.socketOnData (_http_client.js:320:20)
  at emitOne (events.js:77:13)
  at TLSSocket.emit (events.js:169:7)
  at readableAddChunk (_stream_readable.js:153:18)
  at TLSSocket.Readable.push (_stream_readable.js:111:10)
  at TLSWrap.onread (net.js:536:20)
```

Thanks

**Comment by JohnWelborn** [comment]
@emedvedev

**Comment by JohnWelborn** [comment]
I made a workaround for now in post_data.js on line 56 since I don't have large messages.

```
  if (split_message.text) {
    this.robot.messageRoom.call(this.robot, recipient, pretext + split_message.pretext + "\n>>>" + split_message.text);
```

**Comment by emedvedev** [comment]
Thanks for reporting the issue!

> When I install hubot following the instructions on https://github.com/StackStorm/hubot-slack/ 

I'm assuming you meant `hubot-stackstorm`. :) Will make sure to look into it in the next few days.

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/community/p1474048794001641

**Comment by emedvedev** [comment]
@JohnWelborn so, the version of `hubot-slack` that's supposed to work is `3.4.2` (not our fork, the main branch at the "slackhq" org). The current is `4.x.x`, and I assume something broke backward compatibility, that's why external Hubot doesn't work. Since we also need to address https://github.com/StackStorm/st2chatops/issues/56, I'll look into updating to `4.x.x` next week.

**Comment by emedvedev** [comment]
https://github.com/StackStorm/st2chatops/blob/master/package.json#L21
https://github.com/StackStorm/st2chatops/blob/master/npm-shrinkwrap.json#L327

**Comment by armab** [comment]
This is updated/fixed since StackStorm (`st2chatops`) `v2.4.0`
